<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns1="http://www.tibco.com/pe/DeployedVarsType" xmlns:ns="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:pfx2="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:pfx="http://www.tibco.com/CommonLE2/Private/Exception">
    <xs:import xmlns:xs="http://www.w3.org/2001/XMLSchema" namespace="http://www.tibco.com/CommonLE2/Private/Exception" schemaLocation="/SharedResources/SchemaDefinitions/Private/Exception/QueryUpdate.xsd"/>
    <xs:import xmlns:xs="http://www.w3.org/2001/XMLSchema" namespace="http://www.tibco.com/CommonLE2/namespace/private/Common" schemaLocation="/SharedResources/SchemaDefinitions/Private/Common/LocalExceptions.xsd"/>
    <pd:name>ExceptionManagement/DataAccess/Oracle/UpdateExceptionStatus.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType ref="pfx:ExceptionStatusUpdate"/>
    <pd:startX>150</pd:startX>
    <pd:startY>64</pd:startY>
    <pd:returnBindings/>
    <pd:endName>End</pd:endName>
    <pd:endX>439</pd:endX>
    <pd:endY>64</pd:endY>
    <pd:errorSchemas>
        <NotUpdated ref="pfx2:ExceptionRecordNotFound"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1287552751102</pd:targetNamespace>
    <pd:activity name="Update Exception Record Status">
        <pd:type>com.tibco.plugin.jdbc.JDBCUpdateActivity</pd:type>
        <pd:resourceType>ae.activities.JDBCUpdateActivity</pd:resourceType>
        <pd:x>286</pd:x>
        <pd:y>72</pd:y>
        <config>
            <timeout>%%Database/MIG_Timeout%%</timeout>
            <commit>true</commit>
            <emptyStrAsNil>false</emptyStrAsNil>
            <jdbcSharedConfig>/SharedResources/Connections/Database/DBServerOracle.sharedjdbc</jdbcSharedConfig>
            <statement>Update ExceptionRec set status = ? where ExceptionID =?</statement>
            <Prepared_Param_DataType>
                <parameter>
                    <parameterName>Status</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
                <parameter>
                    <parameterName>ExceptionID</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
            </Prepared_Param_DataType>
        </config>
        <pd:inputBindings>
            <jdbcUpdateActivityInput>
                <Status>
                    <xsl:value-of select="$Start/pfx:ExceptionStatusUpdate/pfx:Status"/>
                </Status>
                <ExceptionID>
                    <xsl:value-of select="$Start/pfx:ExceptionStatusUpdate/pfx:ExceptionID"/>
                </ExceptionID>
            </jdbcUpdateActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Generate Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>288</pd:x>
        <pd:y>205</pd:y>
        <config>
            <faultName>NotUpdated</faultName>
        </config>
        <pd:inputBindings>
            <ns:ActivityInput>
                <message>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Exception/Message/ExceptionRecordNotFound"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Exception/Code/ExceptionRecordNotFound"/>
                </messageCode>
                <data>
                    <pfx2:ExceptionRecordNotFound>
                        <pfx2:Message>
                            <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Exception/Message/ExceptionRecordNotFound"/>
                        </pfx2:Message>
                        <pfx2:Code>
                            <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/Exception/Code/ExceptionRecordNotFound"/>
                        </pfx2:Code>
                    </pfx2:ExceptionRecordNotFound>
                </data>
            </ns:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Update Exception Record Status</pd:from>
        <pd:to>End</pd:to>
        <pd:xpathDescription>Updated</pd:xpathDescription>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>xpath</pd:conditionType>
        <pd:xpath>$Update-Exception-Record-Status/jdbcUpdateActivityOutput/noOfUpdates&gt;0</pd:xpath>
    </pd:transition>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Update Exception Record Status</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Update Exception Record Status</pd:from>
        <pd:to>Generate Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>otherwise</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>